torchtune/utils/_device.py

```diff
@@ -156,3 +179,63 @@ def batch_to_device(batch: dict, device: torch.device) -> None:
                 f"""To use batch_to_device, all elements in the batch must be a dict or Tensor.
 Got key "{k}" with value of type {type(v)}"""
             )
+
+
+class DeviceSupport(Enum):
+    """
+    This is a simple enum for compute devices,
+    This currently only supports CPU, CUDA, NPU.
+    The following enumeration defines various device configurations with attributes:
+    1. `device_type` (str): The type of device (e.g., "cpu", "cuda", "npu").
+    2. `device_name` (str): A user-friendly name for the device (e.g., "CPU", "GPU", "NPU").
+    3. `communication_backend` (str): Specifies the backend used for communication on this device (e.g., "gloo", "nccl", "hccl").
+    """
+
+    CPU = ("cpu", "CPU", "gloo")
+    CUDA = ("cuda", "GPU", "nccl")
+    NPU = ("npu", "NPU", "hccl")
+
+    def __init__(
+        self,
+        device_type: str,
+        device_name: str,
+        communication_backend: str,
+    ):
+        self.device_type = device_type
+        self.device_name = device_name
+        self.communication_backend = communication_backend
+
+    @staticmethod
+    def from_type(device_type: str):
+        for member in DeviceSupport:
+            if member.device_type == device_type:
+                return member
+        raise ValueError(f"Unknown device type: {device_type}.")
+
+
+def get_device_support() -> DeviceSupport:
+    """function that gets the DeviceSupport with compute devices based on the current machine.
+
+    This currently only supports CPU, CUDA, NPU.
+
+    Returns:
+        device_support: DeviceSupport
+    """
+    device_type = _get_device_type_from_env()
+    return DeviceSupport.from_type(device_type)
+
+
+def get_torch_device() -> any:
+    """Return the corresponding torch attribute based on the device type string.
+
+    Returns:
+        module: The corresponding torch module, or torch.cuda if not found.
```

Comment:
Kinda related to my above comment (specifically it's why I suggested namespace in the function name instead of module).. I think module is a pretty overloaded term in PyTorch, and when I see this I think of nn.Module. Even though you're using it correctly, maybe we can say something like this instead to mitigate any potential confusion?
```suggestion
        module: The corresponding torch device namespace, or torch.cuda if not found.
```